addons/binding/org.openhab.binding.exec/README.md

```diff
@@ -60,8 +69,79 @@ Thing exec:command:myscript [command="php ./configurations/scripts/script.php %2
 **demo.items**
 
 ```
-String APCRaw "[%s]" (All) {channel="exec:command:apc:output"}
-Switch APCRunning { channel="exec:command:apc:run"}
-Number APCExitValue {channel="exec:command:apc:exit"}
-DateTime APCLastExecution {channel="exec:command:apc:lastexecution"}
+
+String APCRaw "[%s]" (All) {channel="exec:command:apc:output"} 
+String APCRunning { channel="exec:command:apc:run"}
+String APCExitValue {channel="exec:command:apc:exit"}
+String APCLastExecution {channel="exec:command:apc:lastexecution"}
+```
+
+## Full Example
+Following is an example how to set up an exec Thing, debug it with a rule and set the returned string to an Number Item. 
+
+**demo.things**
+```
+Thing exec:command:yourcommand [ command="<YOUR COMMAND> %2$s"         
+                                 interval=0,
+                                 autorun=true ]
+```
+
+**demo.items**
+```
+Switch YourTrigger
+Number YourNumber "Your Number [%.1f °C]"
+
+// state of the execution, is runnung or finished
+Switch yourcommand { channel="exec:command:yourcommand:run" }
+// Arguments to be placed for '%2$s' in command line
+String yourcommand_Args { channel="exec:command:yourcommand:input"}
+// Output of command line execution 
+String yourcommand_out { channel="exec:command:yourcommand:output" }
+```
+
+**demo.sitemap**
+```
+// Name of file and name of sitemap has to be the same
+sitemap your label="Your Value"
+{
+        Frame {
+            Switch item=YourTrigger
+            Text item=YourNumber
+        }
+}
 ```
+
+**demo.rules**
+```
+rule "Your Execution"
+  when
+     Item YourTrigger received update
+  then
+        if(YourTrigger == ON){
+                yourcommand_Args.sendCommand("Additional Arguments")
+        }else{
+                yourcommand_Args.sendCommand("Other Additional Arguments")
+        }
+
+      // wait for the command to complete
+      // State will be NULL if not used before or ON while command is executed
+      while(yourcommand.state != OFF){
+         Thread::sleep(500)
+      }
+      logInfo("Your command exec", "Resut:" + yourcommand_out.state )
+      
+      // If the returned string is just a number it can be parsed
+      // If not a regex or another transformation can be used
+      YourNumber.postUpdate( 
+            (Integer::parseInt(yourcommand_out.state.toString) as Number ) 
+      ) 
+end
+
+```
+The logging massages can be viewed in the Karaf console have a closer look [in the manual](http://docs.openhab.org/administration/console.html) for more information
+
+## Source
+[OpenHAB 1 Addons wiki](https://github.com/openhab/openhab1-addons/wiki/Raspberry-Pi-System-Temperature)
```

Comment:
Please add a newline before and after each heading in a markdown file, otherwise some of the implementations including the one used by [docs.openhab.org](https://docs.openhab.org) might fail to render properly.